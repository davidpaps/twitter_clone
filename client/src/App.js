import React, { Fragment, useState } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import "./App.css";
import SignIn from "./components/signIn/signIn";
import SignUp from "./components/signUp/signUp";
import Twitter from "./components/twitter/twitter";
import Navigation from "./components/navigation/navigation";
import Error from "./components/error/error";
import Home from "./components/home/home";
import SignOut from "./components/signOut/signOut";
import PrivateRoute from "./components/privateRoute";

const App = () => {
  const [auth, setAuth] = useState(false);

  return (
    <Fragment>
      <Router>
        <Navigation />
        <div className="container">
          <ul>
            <Switch>
              <Route exact path="/" component={Home} />
              <Route exact path="/sign_in" component={SignIn} />
              <Route exact path="/sign_up" component={SignUp} />
              <Route exact path="/sign_out" component={SignOut} />
              <PrivateRoute exact path="/home" component={Twitter} />
              <Route component={Error} />
            </Switch>
          </ul>
        </div>
      </Router>
    </Fragment>
  );
};

export default App;
