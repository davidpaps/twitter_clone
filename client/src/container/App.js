import React, { Fragment, useState } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import "./App.css";
import SignIn from "../components/home/signIn/signIn";
import SignUp from "../components/home/signUp/signUp";
import Twitter from "../components/twitter/twitter";
import Navigation from "../components/navigation/navigation";
import Error from "../components/error/error";
import Home from "../components/home/home";
import SignOut from "../components/navigation/signOut/signOut";
import PrivateRoute from "../components/privateRoute/privateRoute";

const App = () => {
  const [auth, setAuth] = useState(false);
  const [username, setUsername] = useState("");

  return (
    <Fragment>
      <Router>
        <Navigation auth={auth} username={username} />
        <div className="container">
          <ul>
            <Switch>
              <Route exact path="/">
                <Home />
              </Route>
              <Route
                exact
                path="/sign_in"
                render={(props) => (
                  <SignIn {...props} auth={setAuth} username={setUsername} />
                )}
              />
              <Route
                exact
                path="/sign_up"
                render={(props) => (
                  <SignUp {...props} auth={setAuth} username={setUsername} />
                )}
              />
              <Route exact path="/sign_out">
                <SignOut />
              </Route>
              <PrivateRoute
                exact
                path="/home"
                component={Twitter}
                username={username}
                auth={auth}
                setAuth={setAuth}
              />
              <Route>
                <Error />
              </Route>
            </Switch>
          </ul>
        </div>
      </Router>
    </Fragment>
  );
};

export default App;
