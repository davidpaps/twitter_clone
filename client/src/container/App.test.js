import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import Navigation from "../components/navigation/navigation";
import renderer from "react-test-renderer";

describe("App", () => {
  let wrapper;

  beforeEach(() => (wrapper = shallow(<App />)));

  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should render the Navigation component", () => {
    expect(wrapper.containsMatchingElement(<Navigation />)).toEqual(true);
  });

  it("should render a <Fragment />", () => {
    expect(wrapper.find("Fragment").length).toEqual(1);
  });

  it("should render a <BrowserRouter />", () => {
    expect(wrapper.find("BrowserRouter").length).toEqual(1);
  });

  it("should render a <Switch />", () => {
    expect(wrapper.find("Switch").length).toEqual(1);
  });

  it("should render a <Route />", () => {
    expect(wrapper.find("Route").length).toEqual(5);
  });

  it("should rendercorrectly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
