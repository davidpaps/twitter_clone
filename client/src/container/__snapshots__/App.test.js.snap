// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should rendercorrectly 1`] = `
Array [
  <div
    className="Navigation"
  >
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        <img
          alt=""
          className="d-inline-block align-top"
          height="30"
          loading="lazy"
          src="https://vignette.wikia.nocookie.net/logopedia/images/9/99/Twitter_2012.svg/revision/latest/scale-to-width-down/340?cb=20180321032720"
          width="30"
        />
        Twitter
      </a>
      <div
        className="navbar-collapse collapse"
      />
      <div>
        <ul
          className="nav navbar-nav navbar-center"
        >
          <li>
            
          </li>
        </ul>
      </div>
      <div>
        <ul
          className="nav navbar-nav navbar-right"
        >
          <li
            className="nav-item active"
          >
            <a
              className="nav-link"
              href="/sign_up"
            >
              Sign In 
              <span
                className="sr-only"
              >
                (current)
              </span>
              <img
                alt=""
                className="d-inline-block "
                height="20"
                loading="lazy"
                src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTg9NgXjYkbVD1Wb4LzGSv9nYR6bC3Z_B4qWw&usqp=CAU"
                width="20"
              />
            </a>
          </li>
          <li
            className="nav-item active"
          >
            <a
              className="nav-link"
              href="/sign_in"
            >
              Sign Up 
              <span
                className="sr-only"
              >
                (current)
              </span>
              <img
                alt=""
                className="d-inline-block"
                height="20"
                loading="lazy"
                src="https://cdn3.iconfinder.com/data/icons/user-interface-2-9/34/169-512.png"
                width="20"
              />
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  <div
    className="container"
  >
    <ul>
      <div
        className="Home"
      >
        <img
          alt=""
          src="https://images.squarespace-cdn.com/content/v1/5b2a905bec4eb77243b05b3c/1531762494480-KPB3OWDFXVL952KXQ8Z6/ke17ZwdGBToddI8pDm48kPipDBN8EsXn0GX4VYEjDvAUqsxRUqqbr1mOJYKfIPR7LoDQ9mXPOjoJoqy81S2I8N_N4V1vUb5AoIIIbLZhVYy7Mythp_T-mtop-vrsUOmeInPi9iDjx9w8K4ZfjXt2di_anSEgls28Fx9rC-ls48EcpI-gmYcFCPdUS0knNFfOCjLISwBs8eEdxAxTptZAUg/welcome.PNG"
        />
        <a
          href="/sign_up"
          onClick={[Function]}
        >
          <button
            className="btn btn-outline-success btn-sm mr-xl-5 w-100 h-100"
          >
            New User? Create an Account!
          </button>
        </a>
        <br />
        <br />
        <a
          href="/sign_in"
          onClick={[Function]}
        >
          <button
            className="btn btn-outline-info btn-sm mr-xl-5 w-100 h-100"
          >
            Already Have an Account? Click Here to Sign In!
          </button>
        </a>
        <img
          alt=""
          className="side"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/d0/Black_Man_Playing_Guitar_Standing_Cartoon_Vector.svg/1280px-Black_Man_Playing_Guitar_Standing_Cartoon_Vector.svg.png"
        />
        <img
          alt=""
          className="side"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/34/Woman_In_Blue_Holding_Nothing_Cartoon.svg/1280px-Woman_In_Blue_Holding_Nothing_Cartoon.svg.png"
        />
        <img
          alt=""
          className="side"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/f/f2/Man_Looking_at_Phone_Cartoon_Vector.svg/1024px-Man_Looking_at_Phone_Cartoon_Vector.svg.png"
        />
      </div>
    </ul>
  </div>,
]
`;
